### PR TITLE
Add crow cpp implementation

### PR DESCRIPTION
**Pull request by moshen**
Add an implementation for [the cpp crow web framework](https://github.com/ipkn/crow), while this is also based on boost asio, it is somewhat simpler than websocketpp.

Also moves the RapidJSON vendor folder up a level since it is reused in this crow implementation.

### CODE REVIEW

File: cpp/crow/src/shared_mutex.hpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/**
+ * Shared Mutex from:
+ * https://isocpp.org/files/papers/N3659.html
```
Comment (moshen) [inline cpp/crow/src/shared_mutex.hpp]: cited the wrong cpp site... will update

File: cpp/uwebsockets/Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 bin/cpp-uwebsockets-server : src/*
-	g++ -std=c++14 -O2 -Ivendor/uWebSockets/src -Ivendor/rapidjson/include src/*.cpp -L$(shell pwd)/vendor/uWebSockets -lpthread -luWS -Wl,-rpath,$(shell pwd)/vendor/uWebSockets -o bin/cpp-uwebsockets-server
+	g++ -std=c++14 -O2 $(CPPFLAGS) -Ivendor/uWebSockets/src -I../vendor/rapidjson/include src/*.cpp -L$(shell pwd)/vendor/uWebSockets -lpthread -luWS -Wl,-rpath,$(shell pwd)/vendor/uWebSockets -o bin/cpp-uwebsockets-server
```
Comment (moshen) [inline cpp/uwebsockets/Makefile]: Oops, added `CPPFLAGS` because I needed them to build on Mac.  Will remove.

**Comment by moshen** [comment]
Fixed my earlier notes

**Comment by jackc** [comment]
I couldn't get this to merge cleanly with the rapidjson submodule move. I had to remove rapidjson, rebase your commit onto that, then re-add it. Anyway, eventually got it to work -- so this is merged, even though github doesn't recognize it.

Thanks!

**Comment by moshen** [comment]
:beers:

**Pull request closed without merging**